jdatetime/__init__.py

```diff
@@ -577,10 +564,22 @@ def strftime(self, format):
         try:
             if self.hour > 12:
                 format = format.replace("%I", '%02.d' % (self.hour - 12))
-            else:
+            elif self.hour > 0:
                 format = format.replace("%I", '%02.d' % (self.hour))
+            else:
+                format = format.replace("%I", '%02.d' % (12))
+        except:
+            format = format.replace("%I", '12')
+
+        try:
+            if self.hour > 12:
```

Comment:
this could be done in the above `if/else`

Reply:
I thought it would be more clear, the other tokens has been written in similar style.